Framework/PythonInterface/mantid/plots/utility.py

```diff
@@ -212,3 +213,14 @@ def get_single_workspace_log_value(ws_index, *, log_values=None, matrix_ws=None,
             return 0
 
         return log_values[ws_index]
+
+
+def is_colorbar(ax):
+    """
+    Check if an axis is a colorbar. Currently in mantid, checking if the
```

Comment:
Unfortunately I don't think this is True as `pcolormesh` produces a `QuadMesh`.  For example, running the script in https://matplotlib.org/3.1.1/gallery/images_contours_and_fields/pcolormesh_levels.html#sphx-glr-gallery-images-contours-and-fields-pcolormesh-levels-py and adding

```python
all_axes = fig.axes
for ax in all_axes:
    print(any(isinstance(col, QuadMesh) for col in ax.collections))
```

shows that the Axes on the top left is also a `QuadMesh` so from this point of view indistinguishable from the colorbar.
I know this was my idea but maybe we can't make a generic function that is able to say this for certain?
My main concern in my original comment was that a previous change went through to check for not activating grid lines on a colorbar and it seemed like a very similar style of check.

Reply:
My bad, I suggested the `QuadMesh` approach because I thought they were unique to colorbars.

Reply:
Previously the check was `len(ax.lines) == 0 and len(ax.images) == 0` but I pointed out that this would also return true for 3D plots, in which case `len(ax.lines) == 0 and len(ax.images) == 0 and not isinstance(ax, Axes3D)` could work?

Reply:
No problem! It's one of those situations where from matplotlib's point of view they don't care that it looks exactly like one of their other types of plot but we do.

I think the key thing is now to understand why the Axes type changes when the options are applied as noted in my last comment. We will probably have to remove this function as we can't guarantee it's correct everytime but I'd like to make sure, at least in `figuremanager` that the various checks for colorbars in grids etc are consistent.